app/Http/Controllers/Api/VolumeController.php

```diff
@@ -256,18 +276,53 @@ private function copyImages($volume, $copy, $selectedImageIds)
      *
      * @param Volume $volume
      * @param Volume $copy
-     * @param int[] $selectedImageIds
+     * @param int[] $selectedFileIds
+     * @param int[] $selectedLabelIds
      **/
-    private function copyImageAnnotation($volume, $copy, $selectedImageIds)
+    private function copyImageAnnotation($volume, $copy, $selectedFileIds, $selectedLabelIds)
     {
+        // if no image ids specified use all images
+        $selectedFileIds = empty($selectedFileIds) ?
+            $volume->images()->pluck('id')->sortBy('id') : $selectedFileIds;
+
+        $annotationJoinLabel = ImageAnnotation::join('image_annotation_labels',
+            'image_annotation_labels.annotation_id', '=', 'image_annotations.id')
+            ->when(!empty($selectedLabelIds), function ($query) use ($selectedLabelIds) {
+                return $query->whereIn('image_annotation_labels.label_id', $selectedLabelIds);
+            })
+            ->whereIn('image_annotations.image_id', $selectedFileIds);
+
+
+        // use unique ids, because an annotation with multiple labels would be duplicated
+        $usedAnnotationIds = array_unique($annotationJoinLabel
+            ->pluck('image_annotations.id')
+            ->toArray());
+
+        if (empty($selectedLabelIds)) {
+            $imageAnnotationLabelIds = $annotationJoinLabel
+                ->pluck('image_annotation_labels.label_id')
+                ->toArray();
+        } else {
+            $imageAnnotationLabelIds = $selectedLabelIds;
+        }
+
         $chunkSize = 100;
         $newImageIds = $copy->images()->orderBy('id')->pluck('id');
         $volume->images()
+            ->with([
+                'annotations' => fn($q) => $q->whereIn('id', $usedAnnotationIds),
+                'annotations.labels' => fn($q) => $q->whereIn('label_id', $imageAnnotationLabelIds),
```

Comment:
I feel that we also need an `orderBy('id')` here.
```suggestion
                'annotations' => fn($q) => $q->whereIn('id', $usedAnnotationIds)->orderBy('id'),
                'annotations.labels' => fn($q) => $q->whereIn('label_id', $imageAnnotationLabelIds)->orderBy('id'),
```